### PR TITLE
chore: remove unused code

### PR DESCRIPTION
**Pull request by ashfurrow**
The type of this PR is: **TYPE**

This PR resolves [CX-892]

### Description

This PR removes the old Auction view, as well as the abstractions used by that view. It also removes unused code throughout the app, notably the spotlight indexing (which doesn't work and wasn't being called), networking code, some unused CocoaPod dependencies, and unit tests for everything else that was removed.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[CX-892]: https://artsyproduct.atlassian.net/browse/CX-892

### CODE REVIEW

**Comment by ashfurrow** [comment]
Hmm, there's a snapshot test for Live Auctions that's failing intermittently, due to antialiasing issues. I'm going to remove it, it's mostly covered by other tests (and is included in the manual QA script).

File: Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/1_-_Slideshow/ARSlideshowViewController.m [hunk 1]
```diff
@@ -58,13 +55,7 @@ - (BOOL)prefersStatusBarHidden
...
 - (void)runSlideshow
 {
-    // devs get it fast
-    if (self.isDeveloper) {
-        [self performSelector:@selector(nextSlide) withObject:nil afterDelay:0.1];
-
-    } else {
-        [self performSelector:@selector(nextSlide) withObject:nil afterDelay:0.6];
-    }
+    [self performSelector:@selector(nextSlide) withObject:nil afterDelay:0.6];
```
Comment (brainbicycle) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/1_-_Slideshow/ARSlideshowViewController.m:58]: Hah I never realized this was different, agree better to make the same ✨

File: Podfile [hunk 2]
```diff
@@ -55,18 +55,13 @@ target 'Artsy' do
...
   pod 'ORStackView', '2.0.3'
   pod 'UIView+BooleanAnimations'
   pod 'Aerodramus', '2.0.0'
 
-  # Custom CollectionView Layouts
-  pod 'ARCollectionViewMasonryLayout', git: 'https://github.com/ashfurrow/ARCollectionViewMasonryLayout'
-
   # Language Enhancements
-  pod 'KSDeferred'
   pod 'MultiDelegate'
```
Comment (brainbicycle) [inline Podfile:65]: 🎉 ✨

**Review by brainbicycle — APPROVED** [review]
Awesome thanks for doing this 🧹!

File: Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/1_-_Slideshow/ARSlideshowViewController.m (hunk 1 above)
Comment (ashfurrow) [reply Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/1_-_Slideshow/ARSlideshowViewController.m:58]: Yeah, this infrastructure had been removed ([docs were removed here](https://github.com/artsy/eigen/pull/2467)) but there used to be a way for developers to have a key/value file in their home directory for certain developer options. I agree we're better off seeing the same thing as users, but if we did want to do something like this again, using `__DEV__` would be the way to go 👍

**Comment by ashfurrow** [comment]
> Do you want this squashed or merged to preserve commits?

Up to you, I see that they were merged in and for such a large PR, I think that makes sense 👍

**Pull request merged**